scripts/run-screenshot-tests.sh

```
@@ -0,0 +1,441 @@
+        available_memory=$(free -m | awk 'NR==2{printf "%.0f", $7/1024}')
+    elif command -v vm_stat &> /dev/null; then
+        # macOS
+        available_memory=$(vm_stat | awk '/free/ {gsub(/\./, "", $3); printf "%.0f", $3/1024/1024}')
+    fi
+    
+    if [[ $available_memory -lt 4 ]]; then
+        print_warning "Low available memory: ${available_memory}GB (4GB+ recommended)"
+    fi
+    
+    # Check available disk space (at least 10GB)
+    local available_disk=$(df . | awk 'NR==2{printf "%.0f", $4/1024/1024}')
```

Shellcheck (0.10.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)